inference-engine/src/inference_engine/include/openvino/runtime/tensor.hpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the OpenVINO Runtime tensor API
+ *
+ * @file openvino/runtime/tensor.hpp
+ */
+#pragma once
+
+#include "common.hpp"
+#include "ie_api.h"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+namespace InferenceEngine {
+class Blob;
+}  // namespace InferenceEngine
+
+namespace ov {
+namespace runtime {
+
+class InferRequest;
+class RemoteTensor;
+class RemoteContext;
+
+/**
+ * @brief Memory access and interpretation API
+ *
+ * It can throw exceptions safely for the application, where it is properly handled.
+ */
+class INFERENCE_ENGINE_API_CLASS(Tensor) {
+    std::shared_ptr<void> _so;
+    ov::element::Type _type;
+    ov::Shape _shape;
+    std::shared_ptr<ie::Blob> _impl;
```

Comment:
Why do we need `Blob` here?

Reply:
Blob here is used as internal implementation. Just internal memory handle but visible for plugins. Not it is forward declared here.